front-end/my-vite-app/src/code-pad/CodeEditor.jsx

```diff
@@ -1,82 +1,120 @@
+import React from "react";
 import {Editor} from "@monaco-editor/react";
 import {useState, useRef, useEffect} from "react";
+import { useParams, Navigate } from 'react-router-dom';
 import LanguageSelector from "./LanguageSelector";
 import Output from "./Output";
-import { STARTER_CODE, THEME } from "../constants";
-import Copilot from "./Copilot"
-import "./CodePad.css"
-
+import { THEME } from "../constants";
+import Copilot from "./Copilot";
+import "./CodePad.css";
 
 const CodeEditor = ({username}) => {
-    const editorRef = useRef()
-    const [value, setValue] = useState(STARTER_CODE['javascript'])
-    const [language, setLanguage] = useState('javascript')
-    const [version, setVersion] = useState('18.15.0')
-    const [languages, setLanguages] = useState([])
-    const [chat, setChat] = useState('')
+    const {idHash} = useParams();
+    const editorRef = useRef();
+    const [IDE, setIDE] = useState({ code: '//pick a language' })
+    const [value, setValue] = useState('//pick a language');
+    const [language, setLanguage] = useState('');
+    const [version, setVersion] = useState('');
+    const [languages, setLanguages] = useState([]);
+    const [chat, setChat] = useState('');
+    const saveTimeout = useRef(null)
+    const prevValueRef = useRef('')
+
 
     useEffect(() => {
         fetchLanguages();
-      }, []);
+        fetchInteractive();
+    }, []);
+
+    useEffect(() => {
+        if(value !== prevValueRef.current){
```

Comment:
where is prevValueRef being initialized?

Reply:
const prevValueRef = useRef('') on line 21